code/game/machinery/hologram.dm

```diff
@@ -41,313 +37,275 @@ var/const/HOLOPAD_MODE = RANGE_BASED
-		to_chat(user, "<span class='notice'>Establishing forced connection to the holopad in [targetpad.loc.loc]</span>")
-		targetpad.forced = 1
+			var/forced_call = FALSE
+			if(access_heads in I.access) //Special functions for command level people
+				switch(alert(user,"Command level authorization detected. Additional functions available.", "Command Level Menu", "Forced Call", "Regular Call"))
+					if("Forced Call")
+						forced_call = TRUE
+			var/list/holopadlist = list()
+			for(var/obj/machinery/hologram/holopad/H in SSmachinery.processing_machines - src)
+				if(ARE_Z_CONNECTED(H.z, z) && H.operable())
+					holopadlist["[H.holopad_id]"] = H	//Define a list and fill it with the area of every holopad in the world
+			holopadlist = sortAssoc(holopadlist)
+			var/chosen_pad = input(user, "Which holopad would you like to contact?", "Holopad List") as null|anything in holopadlist
+			if(!chosen_pad)
+				return
+			connected_pad = holopadlist[chosen_pad]
+			make_call(connected_pad, user, forced_call)
+
+/obj/machinery/hologram/holopad/proc/make_call(var/obj/machinery/hologram/holopad/connected_pad, var/mob/user, forced_call)
+	connected_pad.last_request = world.time
+	connected_pad.connected_pad = src //This marks the holopad you are making the call from
+	connected_pad.incoming_connection = TRUE
+	playsound(connected_pad.loc, 'sound/machines/chime.ogg', 25, 5)
+	connected_pad.update_icon()
+
+	if(forced_call)
+		connected_pad.audible_message("<b>[src]</b> announces, \"Incoming call with command authorization from [connected_pad.holopad_id].\"")
+		to_chat(user, SPAN_NOTICE("Establishing forced connection to the holopad in [connected_pad.holopad_id]."))
+		connected_pad.forced = TRUE
 		sleep(80)
-		targetpad.take_call(user)
-	else
-		targetpad.audible_message("<b>[src]</b> announces, \"Incoming communications request from [targetpad.sourcepad.loc.loc].\"")
-		to_chat(user, "<span class='notice'>Trying to establish a connection to the holopad in [targetpad.loc.loc]... Please await confirmation from recipient.</span>")
-
-/obj/machinery/hologram/holopad/proc/take_call(mob/living/carbon/user)
-	incoming_connection = 0
-	caller_id.set_machine(src)
-	caller_id.reset_view(src)
-	if(!masters[caller_id]) //If there is no hologram, possibly make one.
-		activate_holocall(caller_id)
-
-/obj/machinery/hologram/holopad/proc/end_call(mob/user)
-	if(caller_id)
-		caller_id.unset_machine()
-		caller_id.reset_view() //Send the caller back to his body
-		clear_holo(null, caller_id) // destroy the hologram
-		caller_id = null
-
-	if(user)
-		user.unset_machine()
-		user.reset_view() //Send the caller back to his body
-		clear_holo(null, user) // destroy the hologram
-		user = null
-
-/obj/machinery/hologram/holopad/proc/activate_holocall(mob/living/carbon/caller_id)
-	if(caller_id)
-		create_holo(0, caller_id)//Create one.
-		visible_message("A holographic image of [caller_id] flicks to life right before your eyes!")
```

Comment:
If I read it right we are also losing the ability to see in chat that the AI's hologram has appeard. That'd be a shame I think?